jwst/associations/lib/dms_base.py

```diff
@@ -803,3 +832,20 @@ def item_getattr(item, attributes, association=None):
         attributes,
         invalid_values=invalid_values
     )
+
+
+def nrsfss_valid_detector(item):
+    """Check that a grating/filter combo can appear on the detector"""
+    try:
+        _, detector = item_getattr(item, ['detector'])
+        _, filter = item_getattr(item, ['filter'])
+        _, grating = item_getattr(item, ['grating'])
+        _, slit = item_getattr(item, ['fxd_slit'])
+    except KeyError:
+        return False
+
+    # Reduce all A slits to just 'a'.
+    if slit != 's200b1':
```

Comment:
Is the slit name guaranteed to be in lower case?

Reply:
yes, the association architecture forces all this. It is encapsulated in the `AssociationPool` class.